app/src/main/java/app/revanced/manager/ui/viewmodel/PermissionViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package app.revanced.manager.ui.viewmodel
+
+import android.app.Activity
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.compose.SavedStateHandleSaveableApi
+import app.revanced.manager.util.permissions.AndroidPermission
+import app.revanced.manager.util.permissions.PermissionHelper
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+
+@OptIn(SavedStateHandleSaveableApi::class)
+class PermissionViewModel(
+    @AndroidPermission val permission: String
+) : ViewModel(), KoinComponent {
```

Comment:
You can probably rename this and get rid of `ViewModel()`. Callers from Composables or Viewmodels can easily get their own coroutine scopes.